src/client/camera.cpp

```diff
@@ -533,7 +533,20 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 	m_wieldnode->setPosition(wield_position);
 	m_wieldnode->setRotation(wield_rotation);
 
-	m_wieldnode->setColor(player->light_color);
+	ItemStack wielditemstack;
+	ItemDefinition wielditemdef;
+	IItemDefManager *idef = m_client->getItemDefManager();
+
+	wielditemstack = player->getWieldedItem(&wielditemstack, NULL);
+	wielditemdef = wielditemstack.getDefinition(idef);
```

Comment:
```cpp
    ItemStack wielditem = player->getWieldedItem(&wielditem, nullptr);
    ItemDefinition wielddef = m_client->getItemDefManager.getDefinition(wielditem);
```

Reply:
The first line there causes a segfault. If I separate the declaration and assignment I don't get it anymore.
My code now:
```C++
ItemStack wielditem;
wielditem = lplayer->getWieldedItem(&wielditem, nullptr);
ItemDefinition wielddef = wielditem.getDefinition(m_client->getItemDefManager());
```

Reply:
I'll let you close this if it's resolved